gips/data/gpm/gpm.py

```diff
@@ -0,0 +1,335 @@
+            '_geotransform': (-179.9499969, 0.10000000149011612, 0.0, 89.9499969, 0.0, -0.10000000149011612),
+        },
+        'paccnrtgpmearly': {
+            'description': 'NRT Precipitation Accumulated over 1 Day ending at 06:59 UTC in mm - GPM (and friends) Archive',
+            # the list of asset types associated with this product
+            'assets': ['IMERG-DAY-EARLY'],
+            'startdate': datetime.date(2014, 3, 12),
+            'sensor': 'GPM',
+            '_geotransform': (-179.9499969, 0.10000000149011612, 0.0, 89.9499969, 0.0, -0.10000000149011612),
+        },
+        'paccnrttrmm': {
+            'description': 'NRT Precipitation Accumulated over 1 Day in mm - TRMM (and friends) Archive',
+            # the list of asset types associated with this product
+            'assets': ['3B42-DAY-LATE'],
+            'startdate': datetime.date(2000, 3, 1),
+            'sensor': 'GPM',
+            '_geotransform': (-179.875, 0.25, 0.0, 59.875, 0.0, -0.25),
+        },
+    }
+
+    def asset_check(self, prod_type):
+        """Is an asset available for the current scene and product?
+
+        Returns the last found asset, or else None, its version, the
+        complete lists of missing and available assets, and lastly, an array
+        of pseudo-filepath strings suitable for consumption by gdal/gippy.
+        """
+        # return values
+        asset = None
+        missingassets = []
+        availassets = []
+        allsds = []
+
+        for asset in self._products[prod_type]['assets']:
+            # many asset types won't be found for the current scene
+            if asset not in self.assets:
+                missingassets.append(asset)
+                continue
+            try:
+                sds = self.assets[asset].datafiles()
+            except Exception as e:
+                utils.report_error(e, 'Error reading datafiles for ' + asset)
+                missingassets.append(asset)
+            else:
+                availassets.append(asset)
+                allsds.extend(sds)
+
+        return asset, missingassets, availassets, allsds
+
+    @Data.proc_temp_dir_manager
+    def process(self, *args, **kwargs):
+        """Produce requested products."""
+        products = super(gpmData, self).process(*args, **kwargs)
+        if len(products) == 0:
+            return
+        # example products.requested:
+        # {'temp8tn': ['temp8tn'], 'clouds': ['clouds'], . . . }
+        # key is only used once far below, and val is only used for val[0].
+        for key, val in products.requested.items():
+            start = datetime.datetime.now()
+            prod_type = val[0]
+            asset, missingassets, availassets, allsds = \
+                self.asset_check(prod_type)
+
+            if not availassets:
+                # some products aren't available for every day but this is trying every day
+                VerboseOut('There are no available assets (%s) on %s for tile %s'
```

Comment:
We don't permit new `VerboseOut` calls; use `verbose_out`.  Also use python-3-ready `"{}".format()`.